amazonec2/amazonec2_utils.bal

```diff
@@ -60,6 +60,9 @@ function generateSignature(http:Request request, string accessKeyId, string secr
 
     request.setHeader(CONTENT_TYPE, APPLICATION_URL_ENCODED);
     request.setHeader(X_AMZ_DATE, amzDateStr);
+    if (securityToken != "") {
```

Comment:
Should `securityToken` be an optional field in the config, and a nillable param here instead? Ideally we should be setting whatever the value the user set right? This may be applicable to the other fields too.

Reply:
If we are using temporary ACCESS_KEY_ID, SECRET_ACCESS_KEY, we should set SECURITY_TOKEN  in the config.
If we are using user credentials, we don't need to set SECURITY_TOKEN  in the config.

Reply:
Yeah so we can make it optional right instead of setting empty string?

Reply:
I have made securityToken as optional in method signature like string? securityToken and added this logic.
if (securityToken is string) {
            request.setHeader(X_AMZ_SECURITY_TOKEN, securityToken);
}

getting this error since the empty string is set in the header.
```
<Response><Errors><Error><Code>AuthFailure</Code><Message>AWS was not able to validate the provided access credentials</Message></Error></Errors><RequestID>ebce5d38-80ac-4ea5-b13d-edd90ef8fe94</RequestID></Response>
```

Reply:
Did you also change it in the record?

What I am suggesting is to have it as an optional field in the record, and the field in the object to also be nillable. 

Then we would get `()` here instead of empty string. 

Let's fix this once we change the record.